### ISSUE TITLE
Automation track improvement ideas

### ISSUE THREAD

**Issue by theguitarplayer** [LMMS/lmms#1959]
Here is an idea to make automation of preset tracks easier to see.  Instead of putting automation tracks in the song editor, I would suggest putting the tracks in the piano roll of a certain preset that the automation is being used for.

![lmms 1](https://cloud.githubusercontent.com/assets/6945072/7159301/30a23a2a-e34f-11e4-9ef3-2e26844211b4.png)

This way, it is easier to see which notes are being automated, instead of having to guess.

**Comment by Spekular** [comment]
Hmm, this might be a duplicate of something, but I'll post some feedback
anyways.

Since just about anything can be automated, not all automations make sense
to have in a piano view. In addition, filter frequency, q, volume, pitch
and more are all possible to automate for every preset, so there could be
quite a few automation tracks to display for one instrument track even if
we limit it to only preset related automation.

I'm not saying that this is a bad idea, but I think a different
implementation than in your mockup might be better (+1 for providing a good
mockup though!). Perhaps an overlay of a user selected automation track
would work?

**Comment by Wallacoloo** [comment]
An overlay (or rather something that lays *behind* the notes) does sound
like a better approach to me.

I'm not familiar with lmms I'm depth, but would this actually be useable in
the B/B editor? Since I think beats & bassline patterns are only shallow
copied, there is no single time at which the pattern is played, so there's
no way to map the automation into the piano roll's time.

I think it works with normal patterns though.

**Comment by Spekular** [comment]
I think it should work the same way for bb tracks, I'm assuming the time of
the pattern would be obtained from the song editor rather than the pattern
itself.

**Comment by theguitarplayer** [comment]
Thanks for the comments. 

"filter frequency, q, volume, pitch
and more are all possible to automate for every preset, so there could be
quite a few automation tracks to display for one instrument track even if
we limit it to only preset related automation."

True.  But that shouldn't be a problem (correct me if I'm wrong).
Also, BBline works the same way as song editor, so automating in bb tracks would be no more easier than how it is now.

**Comment by tresf** [comment]
I like the idea in principal.  Most track-based automation overlays the thumbnails or previews anyway, so I would rather take this a bit further and have the lines drawn right over the patterns in the Song Editor, like most video editing software does for fade in/fade out.

-Tres

**Comment by musikBear** [comment]
yes, but then again not so sure. The mock-up will definitely *not* be sufficient. That is way too condensed, and small, and that lead to the next idea - inside piano-roll, it-self
Emediately that *sounds* as a reasonable idea, but it will not be one. 
Piano-roll would be seriously messy and confusing, and it would put really bad strain on the eyes, resulting in head-aches
But the solution is very simple:
Allow the user to 'import' ghosts of any song-editor-block **into** current automation-editor-window, and you would have the benefit of seeing the the notes positions, as you work with the automation. No PR clutter, and also not mandatory, plus less re-design in lmms.

**Comment by Spekular** [comment]
I don't see the difference between overlaying automation on notes vs
overlaying notes on automation. It's to be assumed that the user will
select which automation track to preview and turn it on and lff, right?

**Comment by Spekular** [comment]
Actually it would be cool if this track overlay were implemented between
all track types, perhaps with a little checkbox and dropdown in the toolbar
to enable/dosable and choose a track. What I think would be extra cool
would be a little "switch" button. Say I were overlaying an automation on a
melody and decide it needed to be modified, I would hit switch and the
piano roll would be replaced with an automation editor with the same size
and location, with that melody selected for the overlay. Essentially it
would swap the foreground/editing and background/ghost/reference.

**Comment by musikBear** [comment]
> It's to be assumed that the user will select which automation track to preview and turn it on and lff, right?
If so, then im fine with it, just *looked* like it was permanently present :p
It would indeed be very informative, and intuitive, so in *that* context big :+1:

**Comment by simonvanderveldt** [comment]
Maybe having automation related to notes (like slides) show in the piano roll will already help a bit?

Also, it's probably a good idea to survey what existing DAW's/sequencers do to get a feel for what works well and what doesn't.

**Comment by miketurn** [comment]
Wow there are some great ideas here, this would be nice to see added some day

**Comment by softrabbit** [comment]
With resizable sections in the window (like the note volume/panning already is), I'd take the original mockup over overlays most of the time. 

Overlays could work when automating things related to pitch/frequency, with the scales aligned. For anything else, I don't see that much of a benefit.

And of course this should let the user select any automation that exists in the same timeframe as the note segment. You might want to sync a resonance sweep on one track to a chord in another.
